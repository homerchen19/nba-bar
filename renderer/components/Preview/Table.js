import React from 'react';
import PropTypes from 'prop-types';
import R from 'ramda';
import { Flex } from 'antd-mobile';
import styled from 'styled-components';

import { colors } from '@styles/theme';

const Wrapper = styled.div`
  display: block;
  flex: 3;
  width: 100%;
  overflow-y: scroll !important;
  border-top: 1px solid ${colors.white};
  border-bottom: 1px solid ${colors.white};
  background: #fff;
`;

const Row = styled(Flex)`
  flex: 1;
  width: 100%;
  padding: 10px 15px;
  border-top: 1px solid ${colors.white};
  border-bottom: 1px solid ${colors.white};
`;

const Column = styled(Flex.Item)`
  margin: 0 !important;
  text-align: ${props => props.align};
`;

const Table = ({ home, visitor }) => (
  <Wrapper>
    {R.map(
      key => (
        <Row key={key} justify="center">
          <Column align="left">
            <p>{home[key]}</p>
          </Column>
          <Column align="center">
            <b>
              <p>
                {key === 'plus_minus'
                  ? '+/-'
                  : key.toUpperCase().replace('_PCT', '%')}
              </p>
            </b>
          </Column>
          <Column align="right">
            <p>{visitor[key]}</p>
          </Column>
        </Row>
      ),
      R.keys(home)
    )}
  </Wrapper>
);

Table.propTypes = {
  home: PropTypes.object.isRequired,
  visitor: PropTypes.object.isRequired,
};

export default Table;
