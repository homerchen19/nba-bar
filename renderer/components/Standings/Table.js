import React from 'react';
import PropTypes from 'prop-types';
import { Flex } from 'antd-mobile';
import { Table as VirtualizedTable } from 'react-virtualized';
import styled from 'styled-components';

import { colors } from '@styles/theme';
import nba from '@utils/nba';

const Wrapper = styled(Flex)`
  width: 100%;
  height: 100%;
  overflow-y: scroll !important;
`;

const HeaderCell = styled.div`
  width: ${props => props.width}px;
  padding: 8px 14px;
  background: ${colors.darkBlue};
  color: #fff;
  font-weight: 600;
  text-align: center;
`;

const Cell = styled.div`
  width: ${props => props.width}px;
  padding: 6px 14px;
  border-bottom: 1px solid ${colors.white};
  background: #fff;
  text-align: ${props => props.align || 'center'};
`;

const Conference = styled.h3`
  padding: 10px 0;
  background: ${colors.blue};
  color: #fff;
  text-align: center;
  text-transform: uppercase;
`;

const headerRowRenderer = ({ className, style }) => (
  <Flex className={className} style={style}>
    <HeaderCell key="Teams" width="75">
      Teams
    </HeaderCell>
    <HeaderCell key="W" width="47">
      W
    </HeaderCell>
    <HeaderCell key="L" width="47">
      L
    </HeaderCell>
    <HeaderCell key="WIN%" width="73">
      WIN%
    </HeaderCell>
    <HeaderCell key="GB" width="59">
      GB
    </HeaderCell>
  </Flex>
);

headerRowRenderer.propTypes = {
  className: PropTypes.object.isRequired,
  style: PropTypes.object.isRequired,
};

const rowRenderer = ({ rowData, style }) => (
  <Flex key={rowData.teamId} style={style}>
    <Cell
      key="name"
      align="left"
      style={{
        color: '#fff',
        background: nba.getTeamBackgroundColor(rowData.name),
        fontWeight: '500',
      }}
      width="75"
    >
      {rowData.name}
    </Cell>
    <Cell key="W" width="47">
      {rowData.win}
    </Cell>
    <Cell key="L" width="47">
      {rowData.loss}
    </Cell>
    <Cell key="WIN %" width="73">
      {rowData.winPct}
    </Cell>
    <Cell key="GB" width="59">
      {rowData.gamesBehind}
    </Cell>
  </Flex>
);

rowRenderer.propTypes = {
  rowData: PropTypes.object.isRequired,
  style: PropTypes.object.isRequired,
};

const renderConferenceTable = (teams, conference) => (
  <div key={conference} style={{ width: '100%' }}>
    <Conference>{conference}</Conference>
    <VirtualizedTable
      width={300}
      height={300}
      headerHeight={32}
      rowHeight={30}
      rowCount={teams.length}
      rowGetter={({ index }) => teams[index]}
      headerRowRenderer={headerRowRenderer}
      rowRenderer={rowRenderer}
    />
  </div>
);

const Table = ({ east, west }) => (
  <Wrapper direction="column" align="center">
    {renderConferenceTable(east, 'east')}
    {renderConferenceTable(west, 'west')}
  </Wrapper>
);

Table.propTypes = {
  east: PropTypes.arrayOf(PropTypes.object).isRequired,
  west: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default Table;
