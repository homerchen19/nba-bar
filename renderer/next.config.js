const withCSS = require('@zeit/next-css');
const withImages = require('next-images');

module.exports = withImages(
  withCSS({
    cssModules: true,
    webpack(config, { isServer }) {
      config.target = 'electron-renderer';

      if (isServer) {
        const antStyles = /antd-mobile\/.*?\/style.*?/;
        const origExternals = [...config.externals];
        config.externals = [
          (context, request, callback) => {
            if (request.match(antStyles)) return callback();
            if (typeof origExternals[0] === 'function') {
              origExternals[0](context, request, callback);
            } else {
              callback();
            }
          },
          ...(typeof origExternals[0] === 'function' ? [] : origExternals),
        ];

        config.module.rules.unshift({
          test: antStyles,
          use: 'null-loader',
        });
      }

      return config;
    },
    exportPathMap() {
      return {
        '/home': { page: '/home' },
        '/live': { page: '/live' },
        '/preview': { page: '/preview' },
        '/scoreboard': { page: '/scoreboard' },
        '/settings': { page: '/settings' },
        '/standings': { page: '/standings' },
      };
    },
  })
);
