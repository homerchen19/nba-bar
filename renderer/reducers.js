import { combineReducers } from 'redux';

import homeReducer from './containers/Home/reducer';
import previewReducer from './containers/Preview/reducer';
import liveReducer from './containers/Live/reducer';
import scoreboardReducer from './containers/Scoreboard/reducer';
import standingsReducer from './containers/Standings/reducer';
import settingsReducer from './containers/Settings/reducer';

const rootReducer = combineReducers({
  home: homeReducer,
  preview: previewReducer,
  live: liveReducer,
  scoreboard: scoreboardReducer,
  standings: standingsReducer,
  settings: settingsReducer,
});

export default rootReducer;
