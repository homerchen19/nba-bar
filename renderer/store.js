import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';

import rootReducer from './reducers';

const middleware = [thunk, logger];

const enhancers = compose(
  applyMiddleware(...middleware),
  typeof window !== 'undefined' &&
    window.devToolsExtension &&
    process.env.NODE_ENV !== 'production'
    ? window.devToolsExtension()
    : f => f
);

export default function configureStore(initialState = {}) {
  const store = createStore(rootReducer, initialState, enhancers);

  return store;
}
