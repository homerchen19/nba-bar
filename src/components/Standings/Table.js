import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import R from 'ramda';
import { Flex } from 'antd-mobile';
import { StickyTable, Row, Cell } from 'react-sticky-table';
import styled from 'styled-components';
import { getMainColor } from 'nba-color';

import { colors } from '../../styles/theme';

const Wrapper = styled(Flex)`
  width: 100%;
  overflow-y: scroll !important;

  ::-webkit-scrollbar {
    display: none;
  }
`;

const Item = styled(Flex.Item)`
  width: 100%;
  margin: 0 !important;
`;

const HeaderCell = styled(Cell)`
  padding: 8px 14px;
  background: ${colors.darkBlue};
  color: #fff;
  font-weight: 600;
  text-align: center;
`;

const StyledCell = styled(Cell)`
  width: ${props => props.width}px;
  padding: 6px 14px;
  border-bottom: 1px solid ${colors.white};
  background: #fff;
  text-align: ${props => props.align || 'center'};
`;

const Conference = styled.h3`
  padding: 10px 0;
  background: ${colors.blue};
  color: #fff;
  text-align: center;
  text-transform: uppercase;
`;

const renderHeaderCell = () => {
  const cells = ['Teams', 'W', 'L', 'WIN%', 'GB'];

  return cells.map(cell => <HeaderCell key={cell}>{cell}</HeaderCell>);
};

const renderConferenceTable = (team, conference) => (
  <Fragment>
    <Conference>{conference}</Conference>
    <StickyTable key={conference} stickyColumnCount={0}>
      <Row key="header">{renderHeaderCell()}</Row>
      {R.map(
        teamData => (
          <Row key={teamData.teamId}>
            <StyledCell
              key="name"
              align="left"
              style={{
                color: '#fff',
                background: getMainColor(teamData.name).hex,
                fontWeight: '500',
              }}
              width="75"
            >
              {teamData.name}
            </StyledCell>
            <StyledCell key="W" width="47">
              {teamData.win}
            </StyledCell>
            <StyledCell key="L" width="47">
              {teamData.loss}
            </StyledCell>
            <StyledCell key="WIN %" width="73">
              {teamData.winPct}
            </StyledCell>
            <StyledCell key="GB" width="59">
              {teamData.gamesBehind}
            </StyledCell>
          </Row>
        ),
        team
      )}
    </StickyTable>
  </Fragment>
);

const Table = ({ east, west }) => (
  <Wrapper direction="column" align="center">
    <Item>{renderConferenceTable(east, 'east')}</Item>
    <Item>{renderConferenceTable(west, 'west')}</Item>
  </Wrapper>
);

Table.propTypes = {
  east: PropTypes.arrayOf(PropTypes.object).isRequired,
  west: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default Table;
