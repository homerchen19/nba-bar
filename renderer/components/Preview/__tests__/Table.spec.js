import React from 'react';
import renderer from 'react-test-renderer';

import Table from '../Table';

it('be defined', () => {
  expect(Table).toBeDefined();
});

it('match snapshot', () => {
  const props = {
    home: {
      gp: 78,
      w: 48,
      l: 30,
      pts: 109.1,
      fg_pct: 0.47,
      fg3_pct: 0.367,
      ft_pct: 0.754,
      oreb: 10.8,
      dreb: 36.4,
      reb: 47.2,
      ast: 26.8,
      blk: 5.1,
      stl: 8.2,
      tov: 16.7,
      pf: 22.2,
      plus_minus: 4,
    },
    visitor: {
      gp: 79,
      w: 49,
      l: 30,
      pts: 110.6,
      fg_pct: 0.477,
      fg3_pct: 0.371,
      ft_pct: 0.781,
      oreb: 8.4,
      dreb: 33.7,
      reb: 42.1,
      ast: 23.4,
      blk: 3.9,
      stl: 7.1,
      tov: 13.7,
      pf: 18.7,
      plus_minus: 1,
    },
  };
  const tree = renderer.create(<Table {...props} />).toJSON();

  expect(tree).toMatchSnapshot();
});
