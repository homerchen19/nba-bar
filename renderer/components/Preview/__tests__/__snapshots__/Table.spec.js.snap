// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className="Table__Wrapper-k865a3-0 jOByoO"
>
  <div
    className="am-flexbox Table__Row-k865a3-1 hYyPjy am-flexbox-justify-center am-flexbox-align-center"
  >
    <div
      align="left"
      className="am-flexbox-item Table__Column-k865a3-2 iOtFwu"
    >
      <p>
        78
      </p>
    </div>
    <div
      align="center"
      className="am-flexbox-item Table__Column-k865a3-2 gAbqaP"
    >
      <b>
        <p>
          GP
        </p>
      </b>
    </div>
    <div
      align="right"
      className="am-flexbox-item Table__Column-k865a3-2 kNYEkj"
    >
      <p>
        79
      </p>
    </div>
  </div>
  <div
    className="am-flexbox Table__Row-k865a3-1 hYyPjy am-flexbox-justify-center am-flexbox-align-center"
  >
    <div
      align="left"
      className="am-flexbox-item Table__Column-k865a3-2 iOtFwu"
    >
      <p>
        48
      </p>
    </div>
    <div
      align="center"
      className="am-flexbox-item Table__Column-k865a3-2 gAbqaP"
    >
      <b>
        <p>
          W
        </p>
      </b>
    </div>
    <div
      align="right"
      className="am-flexbox-item Table__Column-k865a3-2 kNYEkj"
    >
      <p>
        49
      </p>
    </div>
  </div>
  <div
    className="am-flexbox Table__Row-k865a3-1 hYyPjy am-flexbox-justify-center am-flexbox-align-center"
  >
    <div
      align="left"
      className="am-flexbox-item Table__Column-k865a3-2 iOtFwu"
    >
      <p>
        30
      </p>
    </div>
    <div
      align="center"
      className="am-flexbox-item Table__Column-k865a3-2 gAbqaP"
    >
      <b>
        <p>
          L
        </p>
      </b>
    </div>
    <div
      align="right"
      className="am-flexbox-item Table__Column-k865a3-2 kNYEkj"
    >
      <p>
        30
      </p>
    </div>
  </div>
  <div
    className="am-flexbox Table__Row-k865a3-1 hYyPjy am-flexbox-justify-center am-flexbox-align-center"
  >
    <div
      align="left"
      className="am-flexbox-item Table__Column-k865a3-2 iOtFwu"
    >
      <p>
        109.1
      </p>
    </div>
    <div
      align="center"
      className="am-flexbox-item Table__Column-k865a3-2 gAbqaP"
    >
      <b>
        <p>
          PTS
        </p>
      </b>
    </div>
    <div
      align="right"
      className="am-flexbox-item Table__Column-k865a3-2 kNYEkj"
    >
      <p>
        110.6
      </p>
    </div>
  </div>
  <div
    className="am-flexbox Table__Row-k865a3-1 hYyPjy am-flexbox-justify-center am-flexbox-align-center"
  >
    <div
      align="left"
      className="am-flexbox-item Table__Column-k865a3-2 iOtFwu"
    >
      <p>
        0.47
      </p>
    </div>
    <div
      align="center"
      className="am-flexbox-item Table__Column-k865a3-2 gAbqaP"
    >
      <b>
        <p>
          FG%
        </p>
      </b>
    </div>
    <div
      align="right"
      className="am-flexbox-item Table__Column-k865a3-2 kNYEkj"
    >
      <p>
        0.477
      </p>
    </div>
  </div>
  <div
    className="am-flexbox Table__Row-k865a3-1 hYyPjy am-flexbox-justify-center am-flexbox-align-center"
  >
    <div
      align="left"
      className="am-flexbox-item Table__Column-k865a3-2 iOtFwu"
    >
      <p>
        0.367
      </p>
    </div>
    <div
      align="center"
      className="am-flexbox-item Table__Column-k865a3-2 gAbqaP"
    >
      <b>
        <p>
          FG3%
        </p>
      </b>
    </div>
    <div
      align="right"
      className="am-flexbox-item Table__Column-k865a3-2 kNYEkj"
    >
      <p>
        0.371
      </p>
    </div>
  </div>
  <div
    className="am-flexbox Table__Row-k865a3-1 hYyPjy am-flexbox-justify-center am-flexbox-align-center"
  >
    <div
      align="left"
      className="am-flexbox-item Table__Column-k865a3-2 iOtFwu"
    >
      <p>
        0.754
      </p>
    </div>
    <div
      align="center"
      className="am-flexbox-item Table__Column-k865a3-2 gAbqaP"
    >
      <b>
        <p>
          FT%
        </p>
      </b>
    </div>
    <div
      align="right"
      className="am-flexbox-item Table__Column-k865a3-2 kNYEkj"
    >
      <p>
        0.781
      </p>
    </div>
  </div>
  <div
    className="am-flexbox Table__Row-k865a3-1 hYyPjy am-flexbox-justify-center am-flexbox-align-center"
  >
    <div
      align="left"
      className="am-flexbox-item Table__Column-k865a3-2 iOtFwu"
    >
      <p>
        10.8
      </p>
    </div>
    <div
      align="center"
      className="am-flexbox-item Table__Column-k865a3-2 gAbqaP"
    >
      <b>
        <p>
          OREB
        </p>
      </b>
    </div>
    <div
      align="right"
      className="am-flexbox-item Table__Column-k865a3-2 kNYEkj"
    >
      <p>
        8.4
      </p>
    </div>
  </div>
  <div
    className="am-flexbox Table__Row-k865a3-1 hYyPjy am-flexbox-justify-center am-flexbox-align-center"
  >
    <div
      align="left"
      className="am-flexbox-item Table__Column-k865a3-2 iOtFwu"
    >
      <p>
        36.4
      </p>
    </div>
    <div
      align="center"
      className="am-flexbox-item Table__Column-k865a3-2 gAbqaP"
    >
      <b>
        <p>
          DREB
        </p>
      </b>
    </div>
    <div
      align="right"
      className="am-flexbox-item Table__Column-k865a3-2 kNYEkj"
    >
      <p>
        33.7
      </p>
    </div>
  </div>
  <div
    className="am-flexbox Table__Row-k865a3-1 hYyPjy am-flexbox-justify-center am-flexbox-align-center"
  >
    <div
      align="left"
      className="am-flexbox-item Table__Column-k865a3-2 iOtFwu"
    >
      <p>
        47.2
      </p>
    </div>
    <div
      align="center"
      className="am-flexbox-item Table__Column-k865a3-2 gAbqaP"
    >
      <b>
        <p>
          REB
        </p>
      </b>
    </div>
    <div
      align="right"
      className="am-flexbox-item Table__Column-k865a3-2 kNYEkj"
    >
      <p>
        42.1
      </p>
    </div>
  </div>
  <div
    className="am-flexbox Table__Row-k865a3-1 hYyPjy am-flexbox-justify-center am-flexbox-align-center"
  >
    <div
      align="left"
      className="am-flexbox-item Table__Column-k865a3-2 iOtFwu"
    >
      <p>
        26.8
      </p>
    </div>
    <div
      align="center"
      className="am-flexbox-item Table__Column-k865a3-2 gAbqaP"
    >
      <b>
        <p>
          AST
        </p>
      </b>
    </div>
    <div
      align="right"
      className="am-flexbox-item Table__Column-k865a3-2 kNYEkj"
    >
      <p>
        23.4
      </p>
    </div>
  </div>
  <div
    className="am-flexbox Table__Row-k865a3-1 hYyPjy am-flexbox-justify-center am-flexbox-align-center"
  >
    <div
      align="left"
      className="am-flexbox-item Table__Column-k865a3-2 iOtFwu"
    >
      <p>
        5.1
      </p>
    </div>
    <div
      align="center"
      className="am-flexbox-item Table__Column-k865a3-2 gAbqaP"
    >
      <b>
        <p>
          BLK
        </p>
      </b>
    </div>
    <div
      align="right"
      className="am-flexbox-item Table__Column-k865a3-2 kNYEkj"
    >
      <p>
        3.9
      </p>
    </div>
  </div>
  <div
    className="am-flexbox Table__Row-k865a3-1 hYyPjy am-flexbox-justify-center am-flexbox-align-center"
  >
    <div
      align="left"
      className="am-flexbox-item Table__Column-k865a3-2 iOtFwu"
    >
      <p>
        8.2
      </p>
    </div>
    <div
      align="center"
      className="am-flexbox-item Table__Column-k865a3-2 gAbqaP"
    >
      <b>
        <p>
          STL
        </p>
      </b>
    </div>
    <div
      align="right"
      className="am-flexbox-item Table__Column-k865a3-2 kNYEkj"
    >
      <p>
        7.1
      </p>
    </div>
  </div>
  <div
    className="am-flexbox Table__Row-k865a3-1 hYyPjy am-flexbox-justify-center am-flexbox-align-center"
  >
    <div
      align="left"
      className="am-flexbox-item Table__Column-k865a3-2 iOtFwu"
    >
      <p>
        16.7
      </p>
    </div>
    <div
      align="center"
      className="am-flexbox-item Table__Column-k865a3-2 gAbqaP"
    >
      <b>
        <p>
          TOV
        </p>
      </b>
    </div>
    <div
      align="right"
      className="am-flexbox-item Table__Column-k865a3-2 kNYEkj"
    >
      <p>
        13.7
      </p>
    </div>
  </div>
  <div
    className="am-flexbox Table__Row-k865a3-1 hYyPjy am-flexbox-justify-center am-flexbox-align-center"
  >
    <div
      align="left"
      className="am-flexbox-item Table__Column-k865a3-2 iOtFwu"
    >
      <p>
        22.2
      </p>
    </div>
    <div
      align="center"
      className="am-flexbox-item Table__Column-k865a3-2 gAbqaP"
    >
      <b>
        <p>
          PF
        </p>
      </b>
    </div>
    <div
      align="right"
      className="am-flexbox-item Table__Column-k865a3-2 kNYEkj"
    >
      <p>
        18.7
      </p>
    </div>
  </div>
  <div
    className="am-flexbox Table__Row-k865a3-1 hYyPjy am-flexbox-justify-center am-flexbox-align-center"
  >
    <div
      align="left"
      className="am-flexbox-item Table__Column-k865a3-2 iOtFwu"
    >
      <p>
        4
      </p>
    </div>
    <div
      align="center"
      className="am-flexbox-item Table__Column-k865a3-2 gAbqaP"
    >
      <b>
        <p>
          +/-
        </p>
      </b>
    </div>
    <div
      align="right"
      className="am-flexbox-item Table__Column-k865a3-2 kNYEkj"
    >
      <p>
        1
      </p>
    </div>
  </div>
</div>
`;
